showcasecard/src/main/java/ru/dimorinny/showcasecard/step/ShowCaseStep.java

```diff
@@ -0,0 +1,116 @@
+package ru.dimorinny.showcasecard.step;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.View;
+
+import ru.dimorinny.showcasecard.position.Center;
+import ru.dimorinny.showcasecard.position.ShowCasePosition;
+import ru.dimorinny.showcasecard.position.ViewPosition;
+
+/**
+ * One tip object, to be shown as a help tip on the screen.
+ */
+public class ShowCaseStep {
+
+    /**
+     * Position on the screen to point the showcase to, when viewToShowCase is not set.
+     */
+    private ShowCasePosition fallbackPosition;
+    /**
+     * Message to display when this item is activated.
+     */
+    private String message;
+    /**
+     * View this showcase is attached to. Null if no view (but a position) is directly attached
+     * to this item.
+     */
+    @Nullable
+    private View viewToShowCase;
```

Comment:
yes, this can be improved. It's been a few weeks, but if I remember correctly, Steps can have two positions: 1. position to scroll to (that centers the view). 2. the position (center of the target view) on the screen to point to after the scroll. ViewPosition only has one coordinate.

Reply:
I'll take a look to fix this